lib/std/x/os/Socket.zig

```diff
@@ -0,0 +1,276 @@
+const std = @import("../../std.zig");
+
+const os = std.os;
+const mem = std.mem;
+const net = std.net;
+const time = std.time;
+const builtin = std.builtin;
+const testing = std.testing;
+
+const Socket = @This();
+
+/// A socket-address pair.
+pub const Connection = struct {
+    socket: Socket,
+    address: net.Address,
```

Comment:
Not sure if it matters a whole lot, but `net.Address` can get pretty big since it's an union with `sockaddr_un`. If I'm just creating a TCP server with this, every connection will needlessly lug around an extra 100 bytes or so.

Reply:
I had started working on this problem [here](https://github.com/LemonBoy/zig/commit/ec622ee5b449b639791129812fe428c3ba173a1f), I hope it can serve as starting point for some further work.

Reply:
> I had started working on this problem [here](https://github.com/LemonBoy/zig/commit/ec622ee5b449b639791129812fe428c3ba173a1f), I hope it can serve as starting point for some further work.

Awesome :smile:! Would you mind if I include it and other codes in a reusable state from your branch in one/more subsequent PR(s)? Also wouldn't mind you taking charge (going to also be PR'ing a few things to do with concurrency after this one).

Reply:
> Would you mind if I include it and other codes in a reusable state from your branch in one/more subsequent PR(s)?

Please do! I've lost steam and left that PR to rot, it'd be great to salvage some code from there.